test/LibraryManager.Test/CacheServiceTest.cs

```diff
@@ -0,0 +1,238 @@
+            CancellationTokenSource tokenSource = new CancellationTokenSource();
+
+            string invalidUrl = "Invalid Url";
+
+            tokenSource.Cancel();
+            await _cacheService.GetCatalogAsync(invalidUrl, _catalogCacheFile, tokenSource.Token);
+
+        }
+
+        [TestMethod]
+        [ExpectedException(typeof(OperationCanceledException))]
+        public async Task HydrateCache_Throws_OperationCanceled_WhenCancelled()
+        {
+            TaskCompletionSource<string> tcs = new TaskCompletionSource<string>();
+            CancellationTokenSource tokenSource = new CancellationTokenSource();
+            string libraryFile1_Path = Path.Combine(_cacheFolder, "Library1", "file1.txt");
+            string libraryFile2_Path = Path.Combine(_cacheFolder, "Library1", "file2.txt");
+            string validUrl = "";
+            string destinationFile = _cacheFolder;
+
+            List<CacheServiceMetadata> desiredCasheFiles = new List<CacheServiceMetadata>()
+            {
+                new CacheServiceMetadata(validUrl, libraryFile1_Path),
+                new CacheServiceMetadata(validUrl, libraryFile2_Path)
+            };
+
+            tokenSource.Cancel();
+            await _cacheService.RefreshCacheAsync(desiredCasheFiles, tokenSource.Token);
+
+        }
+
+        [TestMethod]
+        public async Task HydrateCache_WriteLibraryFilesToCacheFolder()
+        {
+            TaskCompletionSource<string> tcs = new TaskCompletionSource<string>();
+            CancellationTokenSource tokenSource = new CancellationTokenSource();
+            string libraryFile1_Path = Path.Combine(_cacheFolder, "Library1", "file1.txt");
+            string libraryFile2_Path = Path.Combine(_cacheFolder, "Library1", "file2.txt");
+            string validUrl = "";
+            string destinationFile = _cacheFolder;
+
+            List<CacheServiceMetadata> desiredCasheFiles = new List<CacheServiceMetadata>()
+            {
+                new CacheServiceMetadata(validUrl, libraryFile1_Path),
+                new CacheServiceMetadata(validUrl, libraryFile2_Path)
+            };
+
+            // act 
+            await _cacheService.RefreshCacheAsync(desiredCasheFiles, CancellationToken.None);
+
+            // verify
+            Assert.IsTrue(File.Exists(libraryFile1_Path));
+            Assert.IsTrue(File.Exists(libraryFile2_Path));
+        }
+
+        [TestMethod]
+        [Ignore]
+        public async Task CacheAndRestoreMultipleLibraries_WritesAndReadsSuccessfuly()
+        {
+            IEnumerable<CacheServiceMetadata> metadata = GetMetadata(100);
+            await _cacheService.RefreshCacheAsync(metadata, CancellationToken.None);
+            var manifest = Manifest.FromJson(GetLibManConfig(100), _dependencies);
+            
+            // act 
+            IEnumerable<ILibraryInstallationResult> results = await manifest.RestoreAsync(CancellationToken.None).ConfigureAwait(false);
```

Comment:
How is this testing the cache?

Reply:
It is not, that's why I marked as [Ignore]. I will remove it for now until I get it right.